### PR TITLE
Simplify MNIST and NLO challenge via AWS Open Data: Main Branch

### PR DESCRIPTION
**Pull request by rosalieandico**
*Issue #, if available:*
Simplify MNIST and NLO challenge via AWS Open Data
*Description of changes:*
Updated NLP/MNIST exercises.

- local_dir
- decoupled upload to S3 for simplicity and provided description - workshop feedback
- utilized --exclude and --include aws s3 sync options for better file/object management
- re-arranged role,session, bucket assignment immediately before use
- removed util.nb - not needed
- modified util.preprocessing

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by athewsey — CHANGES_REQUESTED** [review]
A few asks to tidy up - I think you'll see the same errors I did if you shut down your studio TensorFlow app, delete all local temporary `data`/`news` folders, maybe consider deleting the corresponding S3 prefixes too, and then try to run through the nbs in order?

File: custom_tensorflow_keras_nlp/Headline Classifier Local.ipynb [hunk 1]
```diff
@@ -25,38 +25,11 @@
...
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "%%time\n",
     "%load_ext autoreload\n",
     "%autoreload 2\n",
```
[the remark this answers is not in the document]
Comment (rosalieandico) [reply custom_tensorflow_keras_nlp/Headline Classifier Local.ipynb]: Done.

File: custom_tensorflow_keras_nlp/Headline Classifier Local.ipynb [hunk 2]
```diff
@@ -74,10 +47,14 @@
...
-    "import util.preprocessing\n",
+    "local_dir = \"news\"\n",
+    "# Download the AG News data from the Registry of Open Data on AWS.\n",
+    "!rm -rf {local_dir}\n",
+    "!mkdir -p {local_dir}\n",
+    "!aws s3 cp s3://fast-ai-nlp/ag_news_csv.tgz {local_dir} --no-sign-request\n",
     "\n",
-    "util.preprocessing.download_dataset()\n"
+    "# Un-tar the AG News data.\n",
+    "!tar zxf {local_dir}/ag_news_csv.tgz -C {local_dir}/ --strip-components=1 --no-same-owner"
```
[the remark this answers is not in the document]
Comment (rosalieandico) [reply custom_tensorflow_keras_nlp/Headline Classifier Local.ipynb:54]: Done.

File: custom_tensorflow_keras_nlp/Headline Classifier Local.ipynb [hunk 3]
```diff
@@ -74,10 +47,14 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "%%time\n",
-    "import util.preprocessing\n",
```
[the remark this answers is not in the document]
Comment (rosalieandico) [reply custom_tensorflow_keras_nlp/Headline Classifier Local.ipynb:78]: Done.

File: custom_tensorflow_keras_nlp/Headline Classifier Local.ipynb [hunk 4]
```diff
@@ -74,10 +47,14 @@
...
+    "local_dir = \"news\"\n",
```
[the remark this answers is not in the document]
Comment (rosalieandico) [reply custom_tensorflow_keras_nlp/Headline Classifier Local.ipynb]: Done.

File: custom_tensorflow_keras_nlp/Headline Classifier SageMaker.ipynb [hunk 5]
```diff
@@ -252,7 +229,7 @@
    "outputs": [],
    "source": [
     "%%time\n",
-    "embedding_matrix = util.preprocessing.get_word_embeddings(tokenizer, \"data/embeddings\")\n"
+    "embedding_matrix = util.preprocessing.get_word_embeddings(tokenizer, f\"{local_dir}/embeddings\")\n"
```
[the remark this answers is not in the document]
Comment (rosalieandico) [reply custom_tensorflow_keras_nlp/Headline Classifier SageMaker.ipynb:233]: Done.

**Review by athewsey — APPROVED** [review]

**Pull request merged**